{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "pediatric-confirmation",
   "metadata": {},
   "source": [
    "As per the ClinVar database link provided in the problem, clinvar.vcf.gz file was downloded.\n",
    "Allele, python library, the vcf file was converted into csv file which can be onened using pandas\n",
    "#allel.vcf_to_csv('clinvar.vcf', 'example1.csv', fields=\"*\")\n",
    "\n",
    "Initially we will import essential python libraries for data processisng."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "recorded-sunday",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "pd.set_option(\"display.max_columns\", None)\n",
    "#pd.set_option(\"display.max_rows\", None)\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "approximate-anthony",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "#using pandas we are reading the csv file which we got from vcf format and storing it into df \n",
    "df = pd.read_csv('example1.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "favorite-investing",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT_1</th>\n",
       "      <th>ALT_2</th>\n",
       "      <th>ALT_3</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>AF_ESP</th>\n",
       "      <th>AF_EXAC</th>\n",
       "      <th>AF_TGP</th>\n",
       "      <th>ALLELEID</th>\n",
       "      <th>CLNDN</th>\n",
       "      <th>CLNDNINCL</th>\n",
       "      <th>CLNDISDB</th>\n",
       "      <th>CLNDISDBINCL</th>\n",
       "      <th>CLNHGVS</th>\n",
       "      <th>CLNREVSTAT</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>CLNSIGCONF</th>\n",
       "      <th>CLNSIGINCL</th>\n",
       "      <th>CLNVC</th>\n",
       "      <th>CLNVCSO</th>\n",
       "      <th>CLNVI</th>\n",
       "      <th>DBVARID</th>\n",
       "      <th>GENEINFO</th>\n",
       "      <th>MC</th>\n",
       "      <th>ORIGIN</th>\n",
       "      <th>RS</th>\n",
       "      <th>SSR</th>\n",
       "      <th>FILTER_PASS</th>\n",
       "      <th>numalt</th>\n",
       "      <th>altlen_1</th>\n",
       "      <th>altlen_2</th>\n",
       "      <th>altlen_3</th>\n",
       "      <th>is_snp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>861332</td>\n",
       "      <td>1019397</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1003021</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MedGen:CN517202</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000001.10:g.861332G&gt;A</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMD11:148398</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>False</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>865519</td>\n",
       "      <td>1125147</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1110865</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MedGen:CN517202</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000001.10:g.865519C&gt;T</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMD11:148398</td>\n",
       "      <td>SO:0001627|intron_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>False</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>865545</td>\n",
       "      <td>1164676</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1153701</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MedGen:CN517202</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000001.10:g.865545G&gt;A</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMD11:148398</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>False</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>865567</td>\n",
       "      <td>1144630</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1131738</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MedGen:CN517202</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000001.10:g.865567C&gt;T</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMD11:148398</td>\n",
       "      <td>SO:0001819|synonymous_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>False</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>865568</td>\n",
       "      <td>846933</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>824438</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MedGen:CN517202</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000001.10:g.865568G&gt;A</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMD11:148398</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>False</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>960101</th>\n",
       "      <td>MT</td>\n",
       "      <td>16011</td>\n",
       "      <td>690273</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>677665</td>\n",
       "      <td>Juvenile_myopathy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MONDO:MONDO:0010789</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_012920.1:m.16011A&gt;G</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MT-TP:4571</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1603225635.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>False</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>960102</th>\n",
       "      <td>MT</td>\n",
       "      <td>16013</td>\n",
       "      <td>690274</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>677666</td>\n",
       "      <td>Juvenile_myopathy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MONDO:MONDO:0010789</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_012920.1:m.16013A&gt;G</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MT-TP:4571</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1603225636.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>False</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>960103</th>\n",
       "      <td>MT</td>\n",
       "      <td>16015</td>\n",
       "      <td>690276</td>\n",
       "      <td>T</td>\n",
       "      <td>TATTCTCTGTTCTTTC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>677667</td>\n",
       "      <td>Juvenile_myopathy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MONDO:MONDO:0010789</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_012920.1:m.16018_16032dup</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Duplication</td>\n",
       "      <td>SO:1000035</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MT-TP:4571</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1603225639.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>False</td>\n",
       "      <td>1</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>960104</th>\n",
       "      <td>MT</td>\n",
       "      <td>16017</td>\n",
       "      <td>690275</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>677668</td>\n",
       "      <td>Juvenile_myopathy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MONDO:MONDO:0010789</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_012920.1:m.16017T&gt;C</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MT-TP:4571</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>201864830.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>False</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>960105</th>\n",
       "      <td>MT</td>\n",
       "      <td>16021</td>\n",
       "      <td>690277</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>677669</td>\n",
       "      <td>Juvenile_myopathy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MONDO:MONDO:0010789</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_012920.1:m.16021C&gt;T</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MT-TP:4571</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1603225641.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>False</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>960106 rows × 36 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM     POS       ID REF             ALT_1  ALT_2  ALT_3  QUAL  \\\n",
       "0          1  861332  1019397   G                 A    NaN    NaN   NaN   \n",
       "1          1  865519  1125147   C                 T    NaN    NaN   NaN   \n",
       "2          1  865545  1164676   G                 A    NaN    NaN   NaN   \n",
       "3          1  865567  1144630   C                 T    NaN    NaN   NaN   \n",
       "4          1  865568   846933   G                 A    NaN    NaN   NaN   \n",
       "...      ...     ...      ...  ..               ...    ...    ...   ...   \n",
       "960101    MT   16011   690273   A                 G    NaN    NaN   NaN   \n",
       "960102    MT   16013   690274   A                 G    NaN    NaN   NaN   \n",
       "960103    MT   16015   690276   T  TATTCTCTGTTCTTTC    NaN    NaN   NaN   \n",
       "960104    MT   16017   690275   T                 C    NaN    NaN   NaN   \n",
       "960105    MT   16021   690277   C                 T    NaN    NaN   NaN   \n",
       "\n",
       "        AF_ESP  AF_EXAC  AF_TGP  ALLELEID              CLNDN CLNDNINCL  \\\n",
       "0          NaN      NaN     NaN   1003021       not_provided       NaN   \n",
       "1          NaN      NaN     NaN   1110865       not_provided       NaN   \n",
       "2          NaN      NaN     NaN   1153701       not_provided       NaN   \n",
       "3          NaN      NaN     NaN   1131738       not_provided       NaN   \n",
       "4          NaN      NaN     NaN    824438       not_provided       NaN   \n",
       "...        ...      ...     ...       ...                ...       ...   \n",
       "960101     NaN      NaN     NaN    677665  Juvenile_myopathy       NaN   \n",
       "960102     NaN      NaN     NaN    677666  Juvenile_myopathy       NaN   \n",
       "960103     NaN      NaN     NaN    677667  Juvenile_myopathy       NaN   \n",
       "960104     NaN      NaN     NaN    677668  Juvenile_myopathy       NaN   \n",
       "960105     NaN      NaN     NaN    677669  Juvenile_myopathy       NaN   \n",
       "\n",
       "                   CLNDISDB CLNDISDBINCL                       CLNHGVS  \\\n",
       "0           MedGen:CN517202          NaN      NC_000001.10:g.861332G>A   \n",
       "1           MedGen:CN517202          NaN      NC_000001.10:g.865519C>T   \n",
       "2           MedGen:CN517202          NaN      NC_000001.10:g.865545G>A   \n",
       "3           MedGen:CN517202          NaN      NC_000001.10:g.865567C>T   \n",
       "4           MedGen:CN517202          NaN      NC_000001.10:g.865568G>A   \n",
       "...                     ...          ...                           ...   \n",
       "960101  MONDO:MONDO:0010789          NaN        NC_012920.1:m.16011A>G   \n",
       "960102  MONDO:MONDO:0010789          NaN        NC_012920.1:m.16013A>G   \n",
       "960103  MONDO:MONDO:0010789          NaN  NC_012920.1:m.16018_16032dup   \n",
       "960104  MONDO:MONDO:0010789          NaN        NC_012920.1:m.16017T>C   \n",
       "960105  MONDO:MONDO:0010789          NaN        NC_012920.1:m.16021C>T   \n",
       "\n",
       "               CLNREVSTAT                  CLNSIG CLNSIGCONF CLNSIGINCL  \\\n",
       "0       criteria_provided  Uncertain_significance        NaN        NaN   \n",
       "1       criteria_provided           Likely_benign        NaN        NaN   \n",
       "2       criteria_provided                  Benign        NaN        NaN   \n",
       "3       criteria_provided           Likely_benign        NaN        NaN   \n",
       "4       criteria_provided  Uncertain_significance        NaN        NaN   \n",
       "...                   ...                     ...        ...        ...   \n",
       "960101  criteria_provided           Likely_benign        NaN        NaN   \n",
       "960102  criteria_provided  Uncertain_significance        NaN        NaN   \n",
       "960103  criteria_provided  Uncertain_significance        NaN        NaN   \n",
       "960104  criteria_provided                  Benign        NaN        NaN   \n",
       "960105  criteria_provided  Uncertain_significance        NaN        NaN   \n",
       "\n",
       "                            CLNVC     CLNVCSO CLNVI DBVARID       GENEINFO  \\\n",
       "0       single_nucleotide_variant  SO:0001483   NaN     NaN  SAMD11:148398   \n",
       "1       single_nucleotide_variant  SO:0001483   NaN     NaN  SAMD11:148398   \n",
       "2       single_nucleotide_variant  SO:0001483   NaN     NaN  SAMD11:148398   \n",
       "3       single_nucleotide_variant  SO:0001483   NaN     NaN  SAMD11:148398   \n",
       "4       single_nucleotide_variant  SO:0001483   NaN     NaN  SAMD11:148398   \n",
       "...                           ...         ...   ...     ...            ...   \n",
       "960101  single_nucleotide_variant  SO:0001483   NaN     NaN     MT-TP:4571   \n",
       "960102  single_nucleotide_variant  SO:0001483   NaN     NaN     MT-TP:4571   \n",
       "960103                Duplication  SO:1000035   NaN     NaN     MT-TP:4571   \n",
       "960104  single_nucleotide_variant  SO:0001483   NaN     NaN     MT-TP:4571   \n",
       "960105  single_nucleotide_variant  SO:0001483   NaN     NaN     MT-TP:4571   \n",
       "\n",
       "                                   MC  ORIGIN            RS  SSR  FILTER_PASS  \\\n",
       "0         SO:0001583|missense_variant     1.0           NaN   -1        False   \n",
       "1           SO:0001627|intron_variant     1.0           NaN   -1        False   \n",
       "2         SO:0001583|missense_variant     1.0           NaN   -1        False   \n",
       "3       SO:0001819|synonymous_variant     1.0           NaN   -1        False   \n",
       "4         SO:0001583|missense_variant     1.0           NaN   -1        False   \n",
       "...                               ...     ...           ...  ...          ...   \n",
       "960101                            NaN     1.0  1603225635.0   -1        False   \n",
       "960102                            NaN     1.0  1603225636.0   -1        False   \n",
       "960103                            NaN     1.0  1603225639.0   -1        False   \n",
       "960104                            NaN     1.0   201864830.0   -1        False   \n",
       "960105                            NaN     1.0  1603225641.0   -1        False   \n",
       "\n",
       "        numalt  altlen_1  altlen_2  altlen_3  is_snp  \n",
       "0            1         0         0         0    True  \n",
       "1            1         0         0         0    True  \n",
       "2            1         0         0         0    True  \n",
       "3            1         0         0         0    True  \n",
       "4            1         0         0         0    True  \n",
       "...        ...       ...       ...       ...     ...  \n",
       "960101       1         0         0         0    True  \n",
       "960102       1         0         0         0    True  \n",
       "960103       1        15         0         0   False  \n",
       "960104       1         0         0         0    True  \n",
       "960105       1         0         0         0    True  \n",
       "\n",
       "[960106 rows x 36 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "instructional-classification",
   "metadata": {},
   "source": [
    "First we will lok at the unique values present in Chromosome."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "apparent-pride",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19,\n",
       "       20, 21, 22, '22', 'X', 'Y', 'MT'], dtype=object)"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.CHROM.unique()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "viral-candle",
   "metadata": {},
   "source": [
    "The 22nd chromosome is mistakenly written as string 22 which can be replaced by 23, since humans have 23 pairs of chromosome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "fantastic-assist",
   "metadata": {},
   "outputs": [],
   "source": [
    "df['CHROM'].replace({'22': 23}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "polished-marks",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19,\n",
       "       20, 21, 22, 23, 'X', 'Y', 'MT'], dtype=object)"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#CHecking if the replace method worked\n",
    "df.CHROM.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "burning-newfoundland",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['ALT_1',\n",
       " 'ALT_2',\n",
       " 'ALT_3',\n",
       " 'QUAL',\n",
       " 'AF_ESP',\n",
       " 'AF_EXAC',\n",
       " 'AF_TGP',\n",
       " 'CLNDN',\n",
       " 'CLNDNINCL',\n",
       " 'CLNDISDB',\n",
       " 'CLNDISDBINCL',\n",
       " 'CLNSIG',\n",
       " 'CLNSIGCONF',\n",
       " 'CLNSIGINCL',\n",
       " 'CLNVI',\n",
       " 'DBVARID',\n",
       " 'GENEINFO',\n",
       " 'MC',\n",
       " 'ORIGIN',\n",
       " 'RS']"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Finding and creating list of features with nan values\n",
    "features_with_na = [features for features in df.columns if df[features].isnull().sum()>1]\n",
    "features_with_na"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "external-photograph",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ALT_1 0.06  % missing\n",
      "ALT_2 100.0  % missing\n",
      "ALT_3 100.0  % missing\n",
      "QUAL 100.0  % missing\n",
      "AF_ESP 84.35000000000001  % missing\n",
      "AF_EXAC 73.81  % missing\n",
      "AF_TGP 81.98  % missing\n",
      "CLNDN 0.06  % missing\n",
      "CLNDNINCL 99.86  % missing\n",
      "CLNDISDB 0.06  % missing\n",
      "CLNDISDBINCL 99.86  % missing\n",
      "CLNSIG 0.06  % missing\n",
      "CLNSIGCONF 94.86  % missing\n",
      "CLNSIGINCL 99.86  % missing\n",
      "CLNVI 76.16000000000001  % missing\n",
      "DBVARID 99.98  % missing\n",
      "GENEINFO 0.03  % missing\n",
      "MC 1.68  % missing\n",
      "ORIGIN 0.5499999999999999  % missing\n",
      "RS 34.23  % missing\n"
     ]
    }
   ],
   "source": [
    "#we will loop through all the features with na value and calculate their nan percentage\n",
    "for feature in features_with_na:\n",
    "    print(feature, (np.round(df[feature].isnull().mean(), 4))*100, ' % missing')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "satellite-workshop",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CHROM               26\n",
       "POS             868268\n",
       "ID              960106\n",
       "REF              13947\n",
       "ALT_1             7382\n",
       "ALT_2                0\n",
       "ALT_3                0\n",
       "QUAL                 0\n",
       "AF_ESP           12923\n",
       "AF_EXAC          13511\n",
       "AF_TGP            5420\n",
       "ALLELEID        960106\n",
       "CLNDN            29448\n",
       "CLNDNINCL          422\n",
       "CLNDISDB          9309\n",
       "CLNDISDBINCL       397\n",
       "CLNHGVS         960106\n",
       "CLNREVSTAT           6\n",
       "CLNSIG              17\n",
       "CLNSIGCONF          61\n",
       "CLNSIGINCL         783\n",
       "CLNVC                8\n",
       "CLNVCSO              8\n",
       "CLNVI            62535\n",
       "DBVARID            153\n",
       "GENEINFO         12939\n",
       "MC                  18\n",
       "ORIGIN              75\n",
       "RS              597843\n",
       "SSR                  1\n",
       "FILTER_PASS          1\n",
       "numalt               2\n",
       "altlen_1          1643\n",
       "altlen_2             1\n",
       "altlen_3             1\n",
       "is_snp               2\n",
       "dtype: int64"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.nunique()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "greek-tunnel",
   "metadata": {},
   "source": [
    "So we are goig to drop all the columns with more than 75% null values and columns with unique values as 1.\n",
    "we will create a list of all the columns and them drop them using drop function in pandas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "linear-embassy",
   "metadata": {},
   "outputs": [],
   "source": [
    "# creating a list of all the columns awith null values more thane 80% and unique values as 1\n",
    "total_col_drop = ['ALT_2', 'ALT_3', 'QUAL', 'AF_ESP', 'AF_EXAC', 'AF_TGP', 'CLNDNINCL', 'CLNDISDBINCL',  'CLNSIGCONF',\n",
    "               'CLNSIGINCL', 'CLNVI', 'DBVARID', 'altlen_3', 'altlen_2', 'SSR', 'FILTER_PASS', 'CLNDISDB', 'CLNHGVS', 'CLNVCSO', 'GENEINFO', 'RS']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "aboriginal-wesley",
   "metadata": {},
   "outputs": [],
   "source": [
    "# dropping all the null values and creating a new dataframe as df1\n",
    "df_1 = df.drop(columns= total_col_drop)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "unauthorized-lender",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT_1</th>\n",
       "      <th>ALLELEID</th>\n",
       "      <th>CLNDN</th>\n",
       "      <th>CLNREVSTAT</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>CLNVC</th>\n",
       "      <th>MC</th>\n",
       "      <th>ORIGIN</th>\n",
       "      <th>numalt</th>\n",
       "      <th>altlen_1</th>\n",
       "      <th>is_snp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>861332</td>\n",
       "      <td>1019397</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1003021</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>865519</td>\n",
       "      <td>1125147</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1110865</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001627|intron_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>865545</td>\n",
       "      <td>1164676</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1153701</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>865567</td>\n",
       "      <td>1144630</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1131738</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001819|synonymous_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>865568</td>\n",
       "      <td>846933</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>824438</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>960101</th>\n",
       "      <td>MT</td>\n",
       "      <td>16011</td>\n",
       "      <td>690273</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>677665</td>\n",
       "      <td>Juvenile_myopathy</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>960102</th>\n",
       "      <td>MT</td>\n",
       "      <td>16013</td>\n",
       "      <td>690274</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>677666</td>\n",
       "      <td>Juvenile_myopathy</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>960103</th>\n",
       "      <td>MT</td>\n",
       "      <td>16015</td>\n",
       "      <td>690276</td>\n",
       "      <td>T</td>\n",
       "      <td>TATTCTCTGTTCTTTC</td>\n",
       "      <td>677667</td>\n",
       "      <td>Juvenile_myopathy</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Duplication</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>15</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>960104</th>\n",
       "      <td>MT</td>\n",
       "      <td>16017</td>\n",
       "      <td>690275</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>677668</td>\n",
       "      <td>Juvenile_myopathy</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>960105</th>\n",
       "      <td>MT</td>\n",
       "      <td>16021</td>\n",
       "      <td>690277</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>677669</td>\n",
       "      <td>Juvenile_myopathy</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>960106 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM     POS       ID REF             ALT_1  ALLELEID  \\\n",
       "0          1  861332  1019397   G                 A   1003021   \n",
       "1          1  865519  1125147   C                 T   1110865   \n",
       "2          1  865545  1164676   G                 A   1153701   \n",
       "3          1  865567  1144630   C                 T   1131738   \n",
       "4          1  865568   846933   G                 A    824438   \n",
       "...      ...     ...      ...  ..               ...       ...   \n",
       "960101    MT   16011   690273   A                 G    677665   \n",
       "960102    MT   16013   690274   A                 G    677666   \n",
       "960103    MT   16015   690276   T  TATTCTCTGTTCTTTC    677667   \n",
       "960104    MT   16017   690275   T                 C    677668   \n",
       "960105    MT   16021   690277   C                 T    677669   \n",
       "\n",
       "                    CLNDN         CLNREVSTAT                  CLNSIG  \\\n",
       "0            not_provided  criteria_provided  Uncertain_significance   \n",
       "1            not_provided  criteria_provided           Likely_benign   \n",
       "2            not_provided  criteria_provided                  Benign   \n",
       "3            not_provided  criteria_provided           Likely_benign   \n",
       "4            not_provided  criteria_provided  Uncertain_significance   \n",
       "...                   ...                ...                     ...   \n",
       "960101  Juvenile_myopathy  criteria_provided           Likely_benign   \n",
       "960102  Juvenile_myopathy  criteria_provided  Uncertain_significance   \n",
       "960103  Juvenile_myopathy  criteria_provided  Uncertain_significance   \n",
       "960104  Juvenile_myopathy  criteria_provided                  Benign   \n",
       "960105  Juvenile_myopathy  criteria_provided  Uncertain_significance   \n",
       "\n",
       "                            CLNVC                             MC  ORIGIN  \\\n",
       "0       single_nucleotide_variant    SO:0001583|missense_variant     1.0   \n",
       "1       single_nucleotide_variant      SO:0001627|intron_variant     1.0   \n",
       "2       single_nucleotide_variant    SO:0001583|missense_variant     1.0   \n",
       "3       single_nucleotide_variant  SO:0001819|synonymous_variant     1.0   \n",
       "4       single_nucleotide_variant    SO:0001583|missense_variant     1.0   \n",
       "...                           ...                            ...     ...   \n",
       "960101  single_nucleotide_variant                            NaN     1.0   \n",
       "960102  single_nucleotide_variant                            NaN     1.0   \n",
       "960103                Duplication                            NaN     1.0   \n",
       "960104  single_nucleotide_variant                            NaN     1.0   \n",
       "960105  single_nucleotide_variant                            NaN     1.0   \n",
       "\n",
       "        numalt  altlen_1  is_snp  \n",
       "0            1         0    True  \n",
       "1            1         0    True  \n",
       "2            1         0    True  \n",
       "3            1         0    True  \n",
       "4            1         0    True  \n",
       "...        ...       ...     ...  \n",
       "960101       1         0    True  \n",
       "960102       1         0    True  \n",
       "960103       1        15   False  \n",
       "960104       1         0    True  \n",
       "960105       1         0    True  \n",
       "\n",
       "[960106 rows x 15 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "historical-growth",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CHROM             0\n",
       "POS               0\n",
       "ID                0\n",
       "REF               0\n",
       "ALT_1           609\n",
       "ALLELEID          0\n",
       "CLNDN           606\n",
       "CLNREVSTAT        0\n",
       "CLNSIG          606\n",
       "CLNVC             0\n",
       "MC            16106\n",
       "ORIGIN         5268\n",
       "numalt            0\n",
       "altlen_1          0\n",
       "is_snp            0\n",
       "dtype: int64"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_1.isnull().sum()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "sitting-strength",
   "metadata": {},
   "source": [
    "df_1 consnsits of 606 null values in CLNDN column and we can also see same numbe of null values are present in CLNSIG column,  so we will check the table and confirm if the null values are present at the same row level."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "virtual-croatia",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT_1</th>\n",
       "      <th>ALLELEID</th>\n",
       "      <th>CLNDN</th>\n",
       "      <th>CLNREVSTAT</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>CLNVC</th>\n",
       "      <th>MC</th>\n",
       "      <th>ORIGIN</th>\n",
       "      <th>numalt</th>\n",
       "      <th>altlen_1</th>\n",
       "      <th>is_snp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5766</th>\n",
       "      <td>1</td>\n",
       "      <td>8021910</td>\n",
       "      <td>487088</td>\n",
       "      <td>G</td>\n",
       "      <td>GGTGCTGGACGGTGTCCCT</td>\n",
       "      <td>75266</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no_interpretation_for_the_single_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Duplication</td>\n",
       "      <td>SO:0001627|intron_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>18</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5805</th>\n",
       "      <td>1</td>\n",
       "      <td>8045031</td>\n",
       "      <td>487089</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>22107</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no_interpretation_for_the_single_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6409</th>\n",
       "      <td>1</td>\n",
       "      <td>10035809</td>\n",
       "      <td>982549</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>970649</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no_interpretation_for_the_single_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001587|nonsense</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8912</th>\n",
       "      <td>1</td>\n",
       "      <td>12008092</td>\n",
       "      <td>242450</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>215016</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no_interpretation_for_the_single_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15282</th>\n",
       "      <td>1</td>\n",
       "      <td>25611101</td>\n",
       "      <td>635094</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>622837</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no_interpretation_for_the_single_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>953331</th>\n",
       "      <td>X</td>\n",
       "      <td>153296212</td>\n",
       "      <td>979165</td>\n",
       "      <td>CTGCTGCGCCCCTTGGGGCTGCTCTCCT</td>\n",
       "      <td>C</td>\n",
       "      <td>967029</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no_interpretation_for_the_single_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>SO:0001822|inframe_deletion</td>\n",
       "      <td>16.0</td>\n",
       "      <td>1</td>\n",
       "      <td>-27</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>953351</th>\n",
       "      <td>X</td>\n",
       "      <td>153296242</td>\n",
       "      <td>1172795</td>\n",
       "      <td>CTTTTCCGCCCAGGGCTCTTACAGGTCT</td>\n",
       "      <td>C</td>\n",
       "      <td>1161881</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no_interpretation_for_the_single_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>SO:0001822|inframe_deletion</td>\n",
       "      <td>16.0</td>\n",
       "      <td>1</td>\n",
       "      <td>-27</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>953377</th>\n",
       "      <td>X</td>\n",
       "      <td>153296290</td>\n",
       "      <td>560201</td>\n",
       "      <td>TCA</td>\n",
       "      <td>T</td>\n",
       "      <td>551241</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no_interpretation_for_the_single_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>SO:0001589|frameshift_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>-2</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>956311</th>\n",
       "      <td>X</td>\n",
       "      <td>153762653</td>\n",
       "      <td>242750</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>38393</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no_interpretation_for_the_single_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>956340</th>\n",
       "      <td>X</td>\n",
       "      <td>153763551</td>\n",
       "      <td>242797</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>25431</td>\n",
       "      <td>NaN</td>\n",
       "      <td>no_interpretation_for_the_single_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>606 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM        POS       ID                           REF  \\\n",
       "5766       1    8021910   487088                             G   \n",
       "5805       1    8045031   487089                             G   \n",
       "6409       1   10035809   982549                             G   \n",
       "8912       1   12008092   242450                             C   \n",
       "15282      1   25611101   635094                             G   \n",
       "...      ...        ...      ...                           ...   \n",
       "953331     X  153296212   979165  CTGCTGCGCCCCTTGGGGCTGCTCTCCT   \n",
       "953351     X  153296242  1172795  CTTTTCCGCCCAGGGCTCTTACAGGTCT   \n",
       "953377     X  153296290   560201                           TCA   \n",
       "956311     X  153762653   242750                             G   \n",
       "956340     X  153763551   242797                             G   \n",
       "\n",
       "                      ALT_1  ALLELEID CLNDN  \\\n",
       "5766    GGTGCTGGACGGTGTCCCT     75266   NaN   \n",
       "5805                      A     22107   NaN   \n",
       "6409                      A    970649   NaN   \n",
       "8912                      T    215016   NaN   \n",
       "15282                     T    622837   NaN   \n",
       "...                     ...       ...   ...   \n",
       "953331                    C    967029   NaN   \n",
       "953351                    C   1161881   NaN   \n",
       "953377                    T    551241   NaN   \n",
       "956311                    A     38393   NaN   \n",
       "956340                    C     25431   NaN   \n",
       "\n",
       "                                      CLNREVSTAT CLNSIG  \\\n",
       "5766    no_interpretation_for_the_single_variant    NaN   \n",
       "5805    no_interpretation_for_the_single_variant    NaN   \n",
       "6409    no_interpretation_for_the_single_variant    NaN   \n",
       "8912    no_interpretation_for_the_single_variant    NaN   \n",
       "15282   no_interpretation_for_the_single_variant    NaN   \n",
       "...                                          ...    ...   \n",
       "953331  no_interpretation_for_the_single_variant    NaN   \n",
       "953351  no_interpretation_for_the_single_variant    NaN   \n",
       "953377  no_interpretation_for_the_single_variant    NaN   \n",
       "956311  no_interpretation_for_the_single_variant    NaN   \n",
       "956340  no_interpretation_for_the_single_variant    NaN   \n",
       "\n",
       "                            CLNVC                             MC  ORIGIN  \\\n",
       "5766                  Duplication      SO:0001627|intron_variant     1.0   \n",
       "5805    single_nucleotide_variant    SO:0001583|missense_variant     1.0   \n",
       "6409    single_nucleotide_variant            SO:0001587|nonsense     1.0   \n",
       "8912    single_nucleotide_variant    SO:0001583|missense_variant     1.0   \n",
       "15282   single_nucleotide_variant    SO:0001583|missense_variant     0.0   \n",
       "...                           ...                            ...     ...   \n",
       "953331                   Deletion    SO:0001822|inframe_deletion    16.0   \n",
       "953351                   Deletion    SO:0001822|inframe_deletion    16.0   \n",
       "953377                   Deletion  SO:0001589|frameshift_variant     1.0   \n",
       "956311  single_nucleotide_variant    SO:0001583|missense_variant     1.0   \n",
       "956340  single_nucleotide_variant    SO:0001583|missense_variant     1.0   \n",
       "\n",
       "        numalt  altlen_1  is_snp  \n",
       "5766         1        18   False  \n",
       "5805         1         0    True  \n",
       "6409         1         0    True  \n",
       "8912         1         0    True  \n",
       "15282        1         0    True  \n",
       "...        ...       ...     ...  \n",
       "953331       1       -27   False  \n",
       "953351       1       -27   False  \n",
       "953377       1        -2   False  \n",
       "956311       1         0    True  \n",
       "956340       1         0    True  \n",
       "\n",
       "[606 rows x 15 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_1[df_1.CLNDN.isnull()]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "exact-doubt",
   "metadata": {},
   "source": [
    "We will use dropna function on CLNDN to remove the null values. as we can see CLNSIG also is null on the same row level"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "cultural-completion",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_1.dropna(subset=['CLNDN'], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "incorporate-mirror",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CHROM             0\n",
       "POS               0\n",
       "ID                0\n",
       "REF               0\n",
       "ALT_1           589\n",
       "ALLELEID          0\n",
       "CLNDN             0\n",
       "CLNREVSTAT        0\n",
       "CLNSIG            0\n",
       "CLNVC             0\n",
       "MC            16092\n",
       "ORIGIN         5247\n",
       "numalt            0\n",
       "altlen_1          0\n",
       "is_snp            0\n",
       "dtype: int64"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_1.isnull().sum()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "viral-georgia",
   "metadata": {},
   "source": [
    "After dropping all the rows from CLNDN column we can see that the rows from CLNSIG also is free of null values."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "graphic-pasta",
   "metadata": {},
   "source": [
    "Since the percent null values in mc and ORIGIN is 1.68% and 0.54% respectively, we will drop the rows. same is the case with ALT_1, which is 0.06%"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "annual-membership",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_1.dropna(subset=['MC'], inplace=True)\n",
    "df_1.dropna(subset=['ORIGIN'], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "western-regression",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_1.dropna(subset=['ALT_1'], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "inappropriate-granny",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT_1</th>\n",
       "      <th>ALLELEID</th>\n",
       "      <th>CLNDN</th>\n",
       "      <th>CLNREVSTAT</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>CLNVC</th>\n",
       "      <th>MC</th>\n",
       "      <th>ORIGIN</th>\n",
       "      <th>numalt</th>\n",
       "      <th>altlen_1</th>\n",
       "      <th>is_snp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>861332</td>\n",
       "      <td>1019397</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1003021</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>865519</td>\n",
       "      <td>1125147</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1110865</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001627|intron_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>865545</td>\n",
       "      <td>1164676</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1153701</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>865567</td>\n",
       "      <td>1144630</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1131738</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001819|synonymous_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>865568</td>\n",
       "      <td>846933</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>824438</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>960022</th>\n",
       "      <td>MT</td>\n",
       "      <td>15873</td>\n",
       "      <td>693971</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>680861</td>\n",
       "      <td>Leigh_syndrome</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>960023</th>\n",
       "      <td>MT</td>\n",
       "      <td>15882</td>\n",
       "      <td>693972</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>680862</td>\n",
       "      <td>Leigh_syndrome</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001587|nonsense</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>960024</th>\n",
       "      <td>MT</td>\n",
       "      <td>15884</td>\n",
       "      <td>143900</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>153625</td>\n",
       "      <td>Neoplasm_of_ovary|Leigh_syndrome</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>960025</th>\n",
       "      <td>MT</td>\n",
       "      <td>15884</td>\n",
       "      <td>252455</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>247210</td>\n",
       "      <td>Leigh_syndrome|not_specified</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>960026</th>\n",
       "      <td>MT</td>\n",
       "      <td>15885</td>\n",
       "      <td>693973</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>680863</td>\n",
       "      <td>Leigh_syndrome</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>937687 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM     POS       ID REF ALT_1  ALLELEID  \\\n",
       "0          1  861332  1019397   G     A   1003021   \n",
       "1          1  865519  1125147   C     T   1110865   \n",
       "2          1  865545  1164676   G     A   1153701   \n",
       "3          1  865567  1144630   C     T   1131738   \n",
       "4          1  865568   846933   G     A    824438   \n",
       "...      ...     ...      ...  ..   ...       ...   \n",
       "960022    MT   15873   693971   T     C    680861   \n",
       "960023    MT   15882   693972   G     A    680862   \n",
       "960024    MT   15884   143900   G     A    153625   \n",
       "960025    MT   15884   252455   G     C    247210   \n",
       "960026    MT   15885   693973   C     T    680863   \n",
       "\n",
       "                                   CLNDN         CLNREVSTAT  \\\n",
       "0                           not_provided  criteria_provided   \n",
       "1                           not_provided  criteria_provided   \n",
       "2                           not_provided  criteria_provided   \n",
       "3                           not_provided  criteria_provided   \n",
       "4                           not_provided  criteria_provided   \n",
       "...                                  ...                ...   \n",
       "960022                    Leigh_syndrome  criteria_provided   \n",
       "960023                    Leigh_syndrome  criteria_provided   \n",
       "960024  Neoplasm_of_ovary|Leigh_syndrome  criteria_provided   \n",
       "960025      Leigh_syndrome|not_specified  criteria_provided   \n",
       "960026                    Leigh_syndrome  criteria_provided   \n",
       "\n",
       "                        CLNSIG                      CLNVC  \\\n",
       "0       Uncertain_significance  single_nucleotide_variant   \n",
       "1                Likely_benign  single_nucleotide_variant   \n",
       "2                       Benign  single_nucleotide_variant   \n",
       "3                Likely_benign  single_nucleotide_variant   \n",
       "4       Uncertain_significance  single_nucleotide_variant   \n",
       "...                        ...                        ...   \n",
       "960022  Uncertain_significance  single_nucleotide_variant   \n",
       "960023  Uncertain_significance  single_nucleotide_variant   \n",
       "960024                  Benign  single_nucleotide_variant   \n",
       "960025                  Benign  single_nucleotide_variant   \n",
       "960026                  Benign  single_nucleotide_variant   \n",
       "\n",
       "                                   MC  ORIGIN  numalt  altlen_1  is_snp  \n",
       "0         SO:0001583|missense_variant     1.0       1         0    True  \n",
       "1           SO:0001627|intron_variant     1.0       1         0    True  \n",
       "2         SO:0001583|missense_variant     1.0       1         0    True  \n",
       "3       SO:0001819|synonymous_variant     1.0       1         0    True  \n",
       "4         SO:0001583|missense_variant     1.0       1         0    True  \n",
       "...                               ...     ...     ...       ...     ...  \n",
       "960022    SO:0001583|missense_variant     1.0       1         0    True  \n",
       "960023            SO:0001587|nonsense     1.0       1         0    True  \n",
       "960024    SO:0001583|missense_variant     1.0       1         0    True  \n",
       "960025    SO:0001583|missense_variant     1.0       1         0    True  \n",
       "960026    SO:0001583|missense_variant     1.0       1         0    True  \n",
       "\n",
       "[937687 rows x 15 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "arctic-retail",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CHROM         0\n",
       "POS           0\n",
       "ID            0\n",
       "REF           0\n",
       "ALT_1         0\n",
       "ALLELEID      0\n",
       "CLNDN         0\n",
       "CLNREVSTAT    0\n",
       "CLNSIG        0\n",
       "CLNVC         0\n",
       "MC            0\n",
       "ORIGIN        0\n",
       "numalt        0\n",
       "altlen_1      0\n",
       "is_snp        0\n",
       "dtype: int64"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_1.isnull().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "premier-sheffield",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "vertical-directory",
   "metadata": {},
   "source": [
    "As per the problem given \n",
    "Classifications are divided into three categories:\n",
    "1. Pathogenic or Likely pathogenic.\n",
    "2. Variants of Uncertain Significance (VUS) and\n",
    "3. Benign or Likely Benign.\n",
    "\n",
    "We will find the value counts of \"CLNSIG\" columns and convert them to comman values like for example 'Likely_benign' will be considered as 'Benign'. \n",
    "drug_response, risk_factor, association, Affects, protective, confers_sensitivity and association_not_found values will be considered as others.\n",
    "So we have 4 values present in this columns which will be easy for our classification model."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "acquired-posting",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Uncertain_significance                          387233\n",
       "Likely_benign                                   235151\n",
       "Benign                                           97020\n",
       "Pathogenic                                       88718\n",
       "Conflicting_interpretations_of_pathogenicity     48453\n",
       "Likely_pathogenic                                40588\n",
       "Benign/Likely_benign                             19233\n",
       "Pathogenic/Likely_pathogenic                      9405\n",
       "not_provided                                      7592\n",
       "drug_response                                     1958\n",
       "other                                             1545\n",
       "risk_factor                                        379\n",
       "association                                        240\n",
       "Affects                                            133\n",
       "protective                                          30\n",
       "confers_sensitivity                                  7\n",
       "association_not_found                                2\n",
       "Name: CLNSIG, dtype: int64"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_1['CLNSIG'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "adapted-champagne",
   "metadata": {},
   "source": [
    "We will create a dictionary of key and values where in keys we will put the value which has to be changed and values will be the value whcih will be asigned. Next setp is to use pandas replace function and use dictionary in its arguments."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "institutional-liver",
   "metadata": {},
   "outputs": [],
   "source": [
    "d ={'Likely_benign': 'Benign', \n",
    "    'Pathogenic/Likely_pathogenic':'Pathogenic',\n",
    "    'Likely_pathogenic': 'Pathogenic',\n",
    "    'Benign/Likely_benign':'Benign', \n",
    "    'not_provided': 'Uncertain_significance',\n",
    "    'Conflicting_interpretations_of_pathogenicity' : 'Uncertain_significance',\n",
    "    'drug_response':'other', \n",
    "    'risk_factor':'other', \n",
    "    'association': 'other', \n",
    "    'Affects':'other', \n",
    "    'protective' : 'other', \n",
    "    'confers_sensitivity': 'other', \n",
    "    'association_not_found' : 'other' }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "frequent-metropolitan",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_1 = df_1.replace({\"CLNSIG\": d})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "elder-skirt",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT_1</th>\n",
       "      <th>ALLELEID</th>\n",
       "      <th>CLNDN</th>\n",
       "      <th>CLNREVSTAT</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>CLNVC</th>\n",
       "      <th>MC</th>\n",
       "      <th>ORIGIN</th>\n",
       "      <th>numalt</th>\n",
       "      <th>altlen_1</th>\n",
       "      <th>is_snp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>861332</td>\n",
       "      <td>1019397</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1003021</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>865519</td>\n",
       "      <td>1125147</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1110865</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001627|intron_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>865545</td>\n",
       "      <td>1164676</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1153701</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>865567</td>\n",
       "      <td>1144630</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1131738</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001819|synonymous_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>865568</td>\n",
       "      <td>846933</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>824438</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>960022</th>\n",
       "      <td>MT</td>\n",
       "      <td>15873</td>\n",
       "      <td>693971</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>680861</td>\n",
       "      <td>Leigh_syndrome</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>960023</th>\n",
       "      <td>MT</td>\n",
       "      <td>15882</td>\n",
       "      <td>693972</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>680862</td>\n",
       "      <td>Leigh_syndrome</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001587|nonsense</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>960024</th>\n",
       "      <td>MT</td>\n",
       "      <td>15884</td>\n",
       "      <td>143900</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>153625</td>\n",
       "      <td>Neoplasm_of_ovary|Leigh_syndrome</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>960025</th>\n",
       "      <td>MT</td>\n",
       "      <td>15884</td>\n",
       "      <td>252455</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>247210</td>\n",
       "      <td>Leigh_syndrome|not_specified</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>960026</th>\n",
       "      <td>MT</td>\n",
       "      <td>15885</td>\n",
       "      <td>693973</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>680863</td>\n",
       "      <td>Leigh_syndrome</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>937687 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM     POS       ID REF ALT_1  ALLELEID  \\\n",
       "0          1  861332  1019397   G     A   1003021   \n",
       "1          1  865519  1125147   C     T   1110865   \n",
       "2          1  865545  1164676   G     A   1153701   \n",
       "3          1  865567  1144630   C     T   1131738   \n",
       "4          1  865568   846933   G     A    824438   \n",
       "...      ...     ...      ...  ..   ...       ...   \n",
       "960022    MT   15873   693971   T     C    680861   \n",
       "960023    MT   15882   693972   G     A    680862   \n",
       "960024    MT   15884   143900   G     A    153625   \n",
       "960025    MT   15884   252455   G     C    247210   \n",
       "960026    MT   15885   693973   C     T    680863   \n",
       "\n",
       "                                   CLNDN         CLNREVSTAT  \\\n",
       "0                           not_provided  criteria_provided   \n",
       "1                           not_provided  criteria_provided   \n",
       "2                           not_provided  criteria_provided   \n",
       "3                           not_provided  criteria_provided   \n",
       "4                           not_provided  criteria_provided   \n",
       "...                                  ...                ...   \n",
       "960022                    Leigh_syndrome  criteria_provided   \n",
       "960023                    Leigh_syndrome  criteria_provided   \n",
       "960024  Neoplasm_of_ovary|Leigh_syndrome  criteria_provided   \n",
       "960025      Leigh_syndrome|not_specified  criteria_provided   \n",
       "960026                    Leigh_syndrome  criteria_provided   \n",
       "\n",
       "                        CLNSIG                      CLNVC  \\\n",
       "0       Uncertain_significance  single_nucleotide_variant   \n",
       "1                       Benign  single_nucleotide_variant   \n",
       "2                       Benign  single_nucleotide_variant   \n",
       "3                       Benign  single_nucleotide_variant   \n",
       "4       Uncertain_significance  single_nucleotide_variant   \n",
       "...                        ...                        ...   \n",
       "960022  Uncertain_significance  single_nucleotide_variant   \n",
       "960023  Uncertain_significance  single_nucleotide_variant   \n",
       "960024                  Benign  single_nucleotide_variant   \n",
       "960025                  Benign  single_nucleotide_variant   \n",
       "960026                  Benign  single_nucleotide_variant   \n",
       "\n",
       "                                   MC  ORIGIN  numalt  altlen_1  is_snp  \n",
       "0         SO:0001583|missense_variant     1.0       1         0    True  \n",
       "1           SO:0001627|intron_variant     1.0       1         0    True  \n",
       "2         SO:0001583|missense_variant     1.0       1         0    True  \n",
       "3       SO:0001819|synonymous_variant     1.0       1         0    True  \n",
       "4         SO:0001583|missense_variant     1.0       1         0    True  \n",
       "...                               ...     ...     ...       ...     ...  \n",
       "960022    SO:0001583|missense_variant     1.0       1         0    True  \n",
       "960023            SO:0001587|nonsense     1.0       1         0    True  \n",
       "960024    SO:0001583|missense_variant     1.0       1         0    True  \n",
       "960025    SO:0001583|missense_variant     1.0       1         0    True  \n",
       "960026    SO:0001583|missense_variant     1.0       1         0    True  \n",
       "\n",
       "[937687 rows x 15 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_1"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "certain-galaxy",
   "metadata": {},
   "source": [
    "We will again check the value counts to ensure if our replacement has taken place successfully"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "plain-tuner",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Uncertain_significance    443278\n",
       "Benign                    351404\n",
       "Pathogenic                138711\n",
       "other                       4294\n",
       "Name: CLNSIG, dtype: int64"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_1.CLNSIG.value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "affecting-bishop",
   "metadata": {},
   "source": [
    "Previously we have done two steps ie handling the null values and columns with unique values as one and working on CLNSIG column.\n",
    "\n",
    "- Next we will focus on the position of each chromosome and check wha the daya tells us\n",
    "\n",
    "* We need to get to every chromosome individually and check for for its positions. Some positions will have only one entry which some may have many entries. For the multiple entries we can see differece in the Alternate and Reference.\n",
    "\n",
    "For our good understanding we will first filter out chromosome 1 and check its positions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "featured-weather",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT_1</th>\n",
       "      <th>ALLELEID</th>\n",
       "      <th>CLNDN</th>\n",
       "      <th>CLNREVSTAT</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>CLNVC</th>\n",
       "      <th>MC</th>\n",
       "      <th>ORIGIN</th>\n",
       "      <th>numalt</th>\n",
       "      <th>altlen_1</th>\n",
       "      <th>is_snp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>861332</td>\n",
       "      <td>1019397</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1003021</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>865519</td>\n",
       "      <td>1125147</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1110865</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001627|intron_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>865545</td>\n",
       "      <td>1164676</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1153701</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>865567</td>\n",
       "      <td>1144630</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1131738</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001819|synonymous_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>865568</td>\n",
       "      <td>846933</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>824438</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76604</th>\n",
       "      <td>1</td>\n",
       "      <td>248616830</td>\n",
       "      <td>916385</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>904391</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001819|synonymous_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76605</th>\n",
       "      <td>1</td>\n",
       "      <td>248651927</td>\n",
       "      <td>403273</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>389409</td>\n",
       "      <td>not_specified</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76606</th>\n",
       "      <td>1</td>\n",
       "      <td>248790108</td>\n",
       "      <td>775702</td>\n",
       "      <td>CAG</td>\n",
       "      <td>C</td>\n",
       "      <td>696649</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>SO:0001587|nonsense</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>-2</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76607</th>\n",
       "      <td>1</td>\n",
       "      <td>249144610</td>\n",
       "      <td>788693</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>707276</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001819|synonymous_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76608</th>\n",
       "      <td>1</td>\n",
       "      <td>249152412</td>\n",
       "      <td>623466</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>612367</td>\n",
       "      <td>Severe_Myopia</td>\n",
       "      <td>no_assertion_criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>74765 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS       ID  REF ALT_1  ALLELEID          CLNDN  \\\n",
       "0         1     861332  1019397    G     A   1003021   not_provided   \n",
       "1         1     865519  1125147    C     T   1110865   not_provided   \n",
       "2         1     865545  1164676    G     A   1153701   not_provided   \n",
       "3         1     865567  1144630    C     T   1131738   not_provided   \n",
       "4         1     865568   846933    G     A    824438   not_provided   \n",
       "...     ...        ...      ...  ...   ...       ...            ...   \n",
       "76604     1  248616830   916385    C     T    904391   not_provided   \n",
       "76605     1  248651927   403273    A     G    389409  not_specified   \n",
       "76606     1  248790108   775702  CAG     C    696649   not_provided   \n",
       "76607     1  249144610   788693    G     A    707276   not_provided   \n",
       "76608     1  249152412   623466    C     G    612367  Severe_Myopia   \n",
       "\n",
       "                           CLNREVSTAT                  CLNSIG  \\\n",
       "0                   criteria_provided  Uncertain_significance   \n",
       "1                   criteria_provided                  Benign   \n",
       "2                   criteria_provided                  Benign   \n",
       "3                   criteria_provided                  Benign   \n",
       "4                   criteria_provided  Uncertain_significance   \n",
       "...                               ...                     ...   \n",
       "76604               criteria_provided                  Benign   \n",
       "76605               criteria_provided                  Benign   \n",
       "76606               criteria_provided                  Benign   \n",
       "76607               criteria_provided                  Benign   \n",
       "76608  no_assertion_criteria_provided  Uncertain_significance   \n",
       "\n",
       "                           CLNVC                             MC  ORIGIN  \\\n",
       "0      single_nucleotide_variant    SO:0001583|missense_variant     1.0   \n",
       "1      single_nucleotide_variant      SO:0001627|intron_variant     1.0   \n",
       "2      single_nucleotide_variant    SO:0001583|missense_variant     1.0   \n",
       "3      single_nucleotide_variant  SO:0001819|synonymous_variant     1.0   \n",
       "4      single_nucleotide_variant    SO:0001583|missense_variant     1.0   \n",
       "...                          ...                            ...     ...   \n",
       "76604  single_nucleotide_variant  SO:0001819|synonymous_variant     1.0   \n",
       "76605  single_nucleotide_variant    SO:0001583|missense_variant     1.0   \n",
       "76606                   Deletion            SO:0001587|nonsense     1.0   \n",
       "76607  single_nucleotide_variant  SO:0001819|synonymous_variant     1.0   \n",
       "76608  single_nucleotide_variant    SO:0001583|missense_variant     0.0   \n",
       "\n",
       "       numalt  altlen_1  is_snp  \n",
       "0           1         0    True  \n",
       "1           1         0    True  \n",
       "2           1         0    True  \n",
       "3           1         0    True  \n",
       "4           1         0    True  \n",
       "...       ...       ...     ...  \n",
       "76604       1         0    True  \n",
       "76605       1         0    True  \n",
       "76606       1        -2   False  \n",
       "76607       1         0    True  \n",
       "76608       1         0    True  \n",
       "\n",
       "[74765 rows x 15 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#we will create a new dataframe which will have data of only one chromosome ie chromosome 1\n",
    "chrom1_df = df_1[df_1['CHROM'] == 1]\n",
    "chrom1_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "configured-syntax",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "92944314     18\n",
       "55524339     11\n",
       "100661987    10\n",
       "20978294     10\n",
       "168262522    10\n",
       "             ..\n",
       "6012770       1\n",
       "35370851      1\n",
       "43224932      1\n",
       "173808690     1\n",
       "220087603     1\n",
       "Name: POS, Length: 70318, dtype: int64"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# we will check the values and their counts of the position\n",
    "chrom1_df['POS'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "experimental-fluid",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT_1</th>\n",
       "      <th>ALLELEID</th>\n",
       "      <th>CLNDN</th>\n",
       "      <th>CLNREVSTAT</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>CLNVC</th>\n",
       "      <th>MC</th>\n",
       "      <th>ORIGIN</th>\n",
       "      <th>numalt</th>\n",
       "      <th>altlen_1</th>\n",
       "      <th>is_snp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>30796</th>\n",
       "      <td>1</td>\n",
       "      <td>92944314</td>\n",
       "      <td>402897</td>\n",
       "      <td>C</td>\n",
       "      <td>CAG</td>\n",
       "      <td>390535</td>\n",
       "      <td>Severe_congenital_neutropenia_2</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>SO:0001627|intron_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30797</th>\n",
       "      <td>1</td>\n",
       "      <td>92944314</td>\n",
       "      <td>298179</td>\n",
       "      <td>C</td>\n",
       "      <td>CAGAG</td>\n",
       "      <td>281299</td>\n",
       "      <td>Severe_congenital_neutropenia|Severe_congenita...</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>SO:0001627|intron_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30798</th>\n",
       "      <td>1</td>\n",
       "      <td>92944314</td>\n",
       "      <td>1170883</td>\n",
       "      <td>C</td>\n",
       "      <td>CAGAGAG</td>\n",
       "      <td>1153696</td>\n",
       "      <td>Severe_congenital_neutropenia_2</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>SO:0001627|intron_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30799</th>\n",
       "      <td>1</td>\n",
       "      <td>92944314</td>\n",
       "      <td>298180</td>\n",
       "      <td>C</td>\n",
       "      <td>CAGAGAGAG</td>\n",
       "      <td>283368</td>\n",
       "      <td>Severe_congenital_neutropenia|Severe_congenita...</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>SO:0001627|intron_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30800</th>\n",
       "      <td>1</td>\n",
       "      <td>92944314</td>\n",
       "      <td>298183</td>\n",
       "      <td>C</td>\n",
       "      <td>CAGAGAGAGAG</td>\n",
       "      <td>281300</td>\n",
       "      <td>Severe_congenital_neutropenia|Severe_congenita...</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>SO:0001627|intron_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30801</th>\n",
       "      <td>1</td>\n",
       "      <td>92944314</td>\n",
       "      <td>770490</td>\n",
       "      <td>C</td>\n",
       "      <td>CAGAGAGAGAGAG</td>\n",
       "      <td>777112</td>\n",
       "      <td>Severe_congenital_neutropenia_2</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>SO:0001627|intron_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30802</th>\n",
       "      <td>1</td>\n",
       "      <td>92944314</td>\n",
       "      <td>298181</td>\n",
       "      <td>C</td>\n",
       "      <td>CAGAGAGAGAGAGAG</td>\n",
       "      <td>283196</td>\n",
       "      <td>Severe_congenital_neutropenia|Severe_congenita...</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>SO:0001627|intron_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>14</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30803</th>\n",
       "      <td>1</td>\n",
       "      <td>92944314</td>\n",
       "      <td>538141</td>\n",
       "      <td>C</td>\n",
       "      <td>CAGAGAGAGAGAGAGAG</td>\n",
       "      <td>516242</td>\n",
       "      <td>Severe_congenital_neutropenia_2</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>SO:0001627|intron_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>16</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30804</th>\n",
       "      <td>1</td>\n",
       "      <td>92944314</td>\n",
       "      <td>788327</td>\n",
       "      <td>C</td>\n",
       "      <td>CAGAGAGAGAGAGAGAGAG</td>\n",
       "      <td>778922</td>\n",
       "      <td>Severe_congenital_neutropenia_2</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>SO:0001627|intron_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>18</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30805</th>\n",
       "      <td>1</td>\n",
       "      <td>92944314</td>\n",
       "      <td>1114439</td>\n",
       "      <td>C</td>\n",
       "      <td>CAGAGAGAGAGAGAGAGAGAG</td>\n",
       "      <td>1089332</td>\n",
       "      <td>Severe_congenital_neutropenia_2</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>SO:0001627|intron_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>20</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30806</th>\n",
       "      <td>1</td>\n",
       "      <td>92944314</td>\n",
       "      <td>800061</td>\n",
       "      <td>C</td>\n",
       "      <td>CAGGGAGAGAG</td>\n",
       "      <td>787070</td>\n",
       "      <td>Severe_congenital_neutropenia_2</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>SO:0001627|intron_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30807</th>\n",
       "      <td>1</td>\n",
       "      <td>92944314</td>\n",
       "      <td>298182</td>\n",
       "      <td>CAG</td>\n",
       "      <td>C</td>\n",
       "      <td>281295</td>\n",
       "      <td>Severe_congenital_neutropenia|Severe_congenita...</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>SO:0001627|intron_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>-2</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30808</th>\n",
       "      <td>1</td>\n",
       "      <td>92944314</td>\n",
       "      <td>298184</td>\n",
       "      <td>CAGAG</td>\n",
       "      <td>C</td>\n",
       "      <td>281937</td>\n",
       "      <td>Severe_congenital_neutropenia|Severe_congenita...</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>SO:0001627|intron_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>-4</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30809</th>\n",
       "      <td>1</td>\n",
       "      <td>92944314</td>\n",
       "      <td>298185</td>\n",
       "      <td>CAGAGAG</td>\n",
       "      <td>C</td>\n",
       "      <td>283353</td>\n",
       "      <td>Severe_congenital_neutropenia|Severe_congenita...</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>SO:0001627|intron_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>-6</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30810</th>\n",
       "      <td>1</td>\n",
       "      <td>92944314</td>\n",
       "      <td>470696</td>\n",
       "      <td>CAGAGAGAG</td>\n",
       "      <td>C</td>\n",
       "      <td>448256</td>\n",
       "      <td>Severe_congenital_neutropenia_2</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>SO:0001627|intron_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>-8</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30811</th>\n",
       "      <td>1</td>\n",
       "      <td>92944314</td>\n",
       "      <td>298186</td>\n",
       "      <td>CAGAGAGAGAG</td>\n",
       "      <td>C</td>\n",
       "      <td>283195</td>\n",
       "      <td>Severe_congenital_neutropenia|Severe_congenita...</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>SO:0001627|intron_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>-10</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30812</th>\n",
       "      <td>1</td>\n",
       "      <td>92944314</td>\n",
       "      <td>722828</td>\n",
       "      <td>CAGAGAGAGAGAG</td>\n",
       "      <td>C</td>\n",
       "      <td>743773</td>\n",
       "      <td>Severe_congenital_neutropenia_2</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>SO:0001627|intron_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>-12</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30813</th>\n",
       "      <td>1</td>\n",
       "      <td>92944314</td>\n",
       "      <td>1128817</td>\n",
       "      <td>CAGAGAGAGAGAGAGAGAGAG</td>\n",
       "      <td>C</td>\n",
       "      <td>1110857</td>\n",
       "      <td>Severe_congenital_neutropenia_2</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>SO:0001627|intron_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>-20</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS       ID                    REF                  ALT_1  \\\n",
       "30796     1  92944314   402897                      C                    CAG   \n",
       "30797     1  92944314   298179                      C                  CAGAG   \n",
       "30798     1  92944314  1170883                      C                CAGAGAG   \n",
       "30799     1  92944314   298180                      C              CAGAGAGAG   \n",
       "30800     1  92944314   298183                      C            CAGAGAGAGAG   \n",
       "30801     1  92944314   770490                      C          CAGAGAGAGAGAG   \n",
       "30802     1  92944314   298181                      C        CAGAGAGAGAGAGAG   \n",
       "30803     1  92944314   538141                      C      CAGAGAGAGAGAGAGAG   \n",
       "30804     1  92944314   788327                      C    CAGAGAGAGAGAGAGAGAG   \n",
       "30805     1  92944314  1114439                      C  CAGAGAGAGAGAGAGAGAGAG   \n",
       "30806     1  92944314   800061                      C            CAGGGAGAGAG   \n",
       "30807     1  92944314   298182                    CAG                      C   \n",
       "30808     1  92944314   298184                  CAGAG                      C   \n",
       "30809     1  92944314   298185                CAGAGAG                      C   \n",
       "30810     1  92944314   470696              CAGAGAGAG                      C   \n",
       "30811     1  92944314   298186            CAGAGAGAGAG                      C   \n",
       "30812     1  92944314   722828          CAGAGAGAGAGAG                      C   \n",
       "30813     1  92944314  1128817  CAGAGAGAGAGAGAGAGAGAG                      C   \n",
       "\n",
       "       ALLELEID                                              CLNDN  \\\n",
       "30796    390535                    Severe_congenital_neutropenia_2   \n",
       "30797    281299  Severe_congenital_neutropenia|Severe_congenita...   \n",
       "30798   1153696                    Severe_congenital_neutropenia_2   \n",
       "30799    283368  Severe_congenital_neutropenia|Severe_congenita...   \n",
       "30800    281300  Severe_congenital_neutropenia|Severe_congenita...   \n",
       "30801    777112                    Severe_congenital_neutropenia_2   \n",
       "30802    283196  Severe_congenital_neutropenia|Severe_congenita...   \n",
       "30803    516242                    Severe_congenital_neutropenia_2   \n",
       "30804    778922                    Severe_congenital_neutropenia_2   \n",
       "30805   1089332                    Severe_congenital_neutropenia_2   \n",
       "30806    787070                    Severe_congenital_neutropenia_2   \n",
       "30807    281295  Severe_congenital_neutropenia|Severe_congenita...   \n",
       "30808    281937  Severe_congenital_neutropenia|Severe_congenita...   \n",
       "30809    283353  Severe_congenital_neutropenia|Severe_congenita...   \n",
       "30810    448256                    Severe_congenital_neutropenia_2   \n",
       "30811    283195  Severe_congenital_neutropenia|Severe_congenita...   \n",
       "30812    743773                    Severe_congenital_neutropenia_2   \n",
       "30813   1110857                    Severe_congenital_neutropenia_2   \n",
       "\n",
       "              CLNREVSTAT                  CLNSIG           CLNVC  \\\n",
       "30796  criteria_provided                  Benign  Microsatellite   \n",
       "30797  criteria_provided  Uncertain_significance  Microsatellite   \n",
       "30798  criteria_provided                  Benign  Microsatellite   \n",
       "30799  criteria_provided  Uncertain_significance  Microsatellite   \n",
       "30800  criteria_provided  Uncertain_significance  Microsatellite   \n",
       "30801  criteria_provided                  Benign  Microsatellite   \n",
       "30802  criteria_provided  Uncertain_significance  Microsatellite   \n",
       "30803  criteria_provided                  Benign  Microsatellite   \n",
       "30804  criteria_provided                  Benign  Microsatellite   \n",
       "30805  criteria_provided                  Benign  Microsatellite   \n",
       "30806  criteria_provided                  Benign       Insertion   \n",
       "30807  criteria_provided  Uncertain_significance  Microsatellite   \n",
       "30808  criteria_provided  Uncertain_significance  Microsatellite   \n",
       "30809  criteria_provided                  Benign  Microsatellite   \n",
       "30810  criteria_provided                  Benign  Microsatellite   \n",
       "30811  criteria_provided  Uncertain_significance  Microsatellite   \n",
       "30812  criteria_provided                  Benign  Microsatellite   \n",
       "30813  criteria_provided                  Benign  Microsatellite   \n",
       "\n",
       "                              MC  ORIGIN  numalt  altlen_1  is_snp  \n",
       "30796  SO:0001627|intron_variant     1.0       1         2   False  \n",
       "30797  SO:0001627|intron_variant     1.0       1         4   False  \n",
       "30798  SO:0001627|intron_variant     1.0       1         6   False  \n",
       "30799  SO:0001627|intron_variant     1.0       1         8   False  \n",
       "30800  SO:0001627|intron_variant     1.0       1        10   False  \n",
       "30801  SO:0001627|intron_variant     1.0       1        12   False  \n",
       "30802  SO:0001627|intron_variant     1.0       1        14   False  \n",
       "30803  SO:0001627|intron_variant     1.0       1        16   False  \n",
       "30804  SO:0001627|intron_variant     1.0       1        18   False  \n",
       "30805  SO:0001627|intron_variant     1.0       1        20   False  \n",
       "30806  SO:0001627|intron_variant     1.0       1        10   False  \n",
       "30807  SO:0001627|intron_variant     1.0       1        -2   False  \n",
       "30808  SO:0001627|intron_variant     1.0       1        -4   False  \n",
       "30809  SO:0001627|intron_variant     1.0       1        -6   False  \n",
       "30810  SO:0001627|intron_variant     1.0       1        -8   False  \n",
       "30811  SO:0001627|intron_variant     1.0       1       -10   False  \n",
       "30812  SO:0001627|intron_variant     1.0       1       -12   False  \n",
       "30813  SO:0001627|intron_variant     1.0       1       -20   False  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chrom1_df[chrom1_df['POS'] == 92944314]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "guilty-hundred",
   "metadata": {},
   "source": [
    "We have filtered chromosome 1 position 92944314, which has 18 entries. if we observe the CLNSIG column we can find different values.  As per the given problem classifications are said to be conflicting if two of the three categories are present for a variant, such case is considered as conflicting. \n",
    "\n",
    "* Like wise we have to go through every position and find if CLNSIG columns have one category of many."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "middle-flower",
   "metadata": {},
   "source": [
    "But initially we will create dataframes with individual chromosomes with multiple entries, since with one variant entry we cannot classify them to be conflicting or not."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "cardiovascular-coordinator",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[1,\n",
       " 2,\n",
       " 3,\n",
       " 4,\n",
       " 5,\n",
       " 6,\n",
       " 7,\n",
       " 8,\n",
       " 9,\n",
       " 10,\n",
       " 11,\n",
       " 12,\n",
       " 13,\n",
       " 14,\n",
       " 15,\n",
       " 16,\n",
       " 17,\n",
       " 18,\n",
       " 19,\n",
       " 20,\n",
       " 21,\n",
       " 22,\n",
       " 23,\n",
       " 'X',\n",
       " 'Y',\n",
       " 'MT']"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Creating a list of individual chromosomes\n",
    "chrom_list = list(df_1.CHROM.unique())\n",
    "chrom_list"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "behind-notice",
   "metadata": {},
   "source": [
    "We need to create a function whcih gives us a dataframe for individual chromosome consisting of positions with multiple variation. Position with single variation will be dropped.\n",
    "\n",
    "So the function works in this way:\n",
    "1. The function takes two arguments 1 a dataframe and the chromosome number\n",
    "2. Depending on the chromosome number the dataframe is filtered out in the next step and stored in 'x' variable.\n",
    "3. in order to identify positions with multiple variation we have to find value counts of 'x', and create a new table using   reset index. (We have a table which has position number and its number of values)\n",
    "4. This new table columns will be replaced as 'POS' and 'num, for our understanding  \n",
    "5. We will filter out num greater than 1, since we want positions with multiple values, and create x_new table form it.\n",
    "6. We will create a 'c' varable and convert the positions to numpy array and store in 'c' variable.\n",
    "7. We will return a chrom dataframe(which will have all the positions which are in variable 'c') by using apply function in pandas."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "wooden-issue",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_chrom(df, chrom_no):\n",
    "    chrom = df[df['CHROM'] == chrom_no] \n",
    "    x = chrom['POS'].value_counts()\n",
    "    x = x.reset_index()\n",
    "    x.columns= ['POS', 'num']\n",
    "    x_new= x.loc[x['num'] > 1]\n",
    "    c = x_new['POS'].to_numpy()\n",
    "    chrom['POS'] = chrom['POS'].apply(lambda x : x if x in c else None)\n",
    "    chrom.dropna(subset = ['POS'], inplace=True)\n",
    "    return chrom"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "regional-scholar",
   "metadata": {},
   "outputs": [],
   "source": [
    "# we will asign the function to c1,  check if its working properly or not\n",
    "c1 = get_chrom(df_1, 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "norwegian-tissue",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT_1</th>\n",
       "      <th>ALLELEID</th>\n",
       "      <th>CLNDN</th>\n",
       "      <th>CLNREVSTAT</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>CLNVC</th>\n",
       "      <th>MC</th>\n",
       "      <th>ORIGIN</th>\n",
       "      <th>numalt</th>\n",
       "      <th>altlen_1</th>\n",
       "      <th>is_snp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>1</td>\n",
       "      <td>871143.0</td>\n",
       "      <td>1127328</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1110870</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001627|intron_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>1</td>\n",
       "      <td>871143.0</td>\n",
       "      <td>1158587</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>1131742</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001627|intron_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>1</td>\n",
       "      <td>871215.0</td>\n",
       "      <td>1170011</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1153710</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001819|synonymous_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>1</td>\n",
       "      <td>871215.0</td>\n",
       "      <td>1169806</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1153711</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001819|synonymous_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>1</td>\n",
       "      <td>874456.0</td>\n",
       "      <td>1122639</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1110873</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76517</th>\n",
       "      <td>1</td>\n",
       "      <td>247599355.0</td>\n",
       "      <td>97960</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>103852</td>\n",
       "      <td>Familial_cold_autoinflammatory_syndrome_1|not_...</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76530</th>\n",
       "      <td>1</td>\n",
       "      <td>247607319.0</td>\n",
       "      <td>1096997</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1067166</td>\n",
       "      <td>Cryopyrin_associated_periodic_syndrome</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001819|synonymous_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76531</th>\n",
       "      <td>1</td>\n",
       "      <td>247607319.0</td>\n",
       "      <td>795624</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>780649</td>\n",
       "      <td>Cryopyrin_associated_periodic_syndrome|not_pro...</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001819|synonymous_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76541</th>\n",
       "      <td>1</td>\n",
       "      <td>247607421.0</td>\n",
       "      <td>1160486</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1131312</td>\n",
       "      <td>Cryopyrin_associated_periodic_syndrome</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001819|synonymous_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76542</th>\n",
       "      <td>1</td>\n",
       "      <td>247607421.0</td>\n",
       "      <td>546637</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>537126</td>\n",
       "      <td>Cryopyrin_associated_periodic_syndrome|not_spe...</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001819|synonymous_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8402 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM          POS       ID REF ALT_1  ALLELEID  \\\n",
       "34        1     871143.0  1127328   G     A   1110870   \n",
       "35        1     871143.0  1158587   G     T   1131742   \n",
       "46        1     871215.0  1170011   C     G   1153710   \n",
       "47        1     871215.0  1169806   C     T   1153711   \n",
       "66        1     874456.0  1122639   G     A   1110873   \n",
       "...     ...          ...      ...  ..   ...       ...   \n",
       "76517     1  247599355.0    97960   A     G    103852   \n",
       "76530     1  247607319.0  1096997   C     G   1067166   \n",
       "76531     1  247607319.0   795624   C     T    780649   \n",
       "76541     1  247607421.0  1160486   C     G   1131312   \n",
       "76542     1  247607421.0   546637   C     T    537126   \n",
       "\n",
       "                                                   CLNDN         CLNREVSTAT  \\\n",
       "34                                          not_provided  criteria_provided   \n",
       "35                                          not_provided  criteria_provided   \n",
       "46                                          not_provided  criteria_provided   \n",
       "47                                          not_provided  criteria_provided   \n",
       "66                                          not_provided  criteria_provided   \n",
       "...                                                  ...                ...   \n",
       "76517  Familial_cold_autoinflammatory_syndrome_1|not_...  criteria_provided   \n",
       "76530             Cryopyrin_associated_periodic_syndrome  criteria_provided   \n",
       "76531  Cryopyrin_associated_periodic_syndrome|not_pro...  criteria_provided   \n",
       "76541             Cryopyrin_associated_periodic_syndrome  criteria_provided   \n",
       "76542  Cryopyrin_associated_periodic_syndrome|not_spe...  criteria_provided   \n",
       "\n",
       "           CLNSIG                      CLNVC                             MC  \\\n",
       "34         Benign  single_nucleotide_variant      SO:0001627|intron_variant   \n",
       "35         Benign  single_nucleotide_variant      SO:0001627|intron_variant   \n",
       "46         Benign  single_nucleotide_variant  SO:0001819|synonymous_variant   \n",
       "47         Benign  single_nucleotide_variant  SO:0001819|synonymous_variant   \n",
       "66         Benign  single_nucleotide_variant    SO:0001583|missense_variant   \n",
       "...           ...                        ...                            ...   \n",
       "76517  Pathogenic  single_nucleotide_variant    SO:0001583|missense_variant   \n",
       "76530      Benign  single_nucleotide_variant  SO:0001819|synonymous_variant   \n",
       "76531      Benign  single_nucleotide_variant  SO:0001819|synonymous_variant   \n",
       "76541      Benign  single_nucleotide_variant  SO:0001819|synonymous_variant   \n",
       "76542      Benign  single_nucleotide_variant  SO:0001819|synonymous_variant   \n",
       "\n",
       "       ORIGIN  numalt  altlen_1  is_snp  \n",
       "34        1.0       1         0    True  \n",
       "35        1.0       1         0    True  \n",
       "46        1.0       1         0    True  \n",
       "47        1.0       1         0    True  \n",
       "66        1.0       1         0    True  \n",
       "...       ...     ...       ...     ...  \n",
       "76517     1.0       1         0    True  \n",
       "76530     1.0       1         0    True  \n",
       "76531     1.0       1         0    True  \n",
       "76541     1.0       1         0    True  \n",
       "76542     1.0       1         0    True  \n",
       "\n",
       "[8402 rows x 15 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "c1"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "unauthorized-rachel",
   "metadata": {},
   "source": [
    "our function worked fine since we can see in dataframe above."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "lyric-idaho",
   "metadata": {},
   "source": [
    "We will also create a function which will add a new column \"CONFLICTING\" which will have 0 and 1, 0 suggesting there is only one category in 'CLNSIG' column and 1 if there are more categories.\n",
    "\n",
    "The function will be as follows:\n",
    "1. The function will take one argument ie a dataframe.\n",
    "1. For the dataframe we created it will add a column \"CONFLICTING\" with 0 value in it.\n",
    "2. We will create a list of Position present in the dataframe, using set function we will remove the duplicates\n",
    "3. We will loop over the position list we created, and filter each and every position individually and check for CLNSIG category\n",
    "4. We will use set function and turn it into len, and if there are multiple categories the length will be more than 1.\n",
    "5. If it is more than 1 then the CONFLICTING value will be converted to 1.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "labeled-healthcare",
   "metadata": {},
   "outputs": [],
   "source": [
    "def conflicting(c):\n",
    "    c['CONFLICTING'] = 0\n",
    "    pos_list = list(set(c.POS))\n",
    "    for i in pos_list: \n",
    "        if len(set(c[c['POS'] == i]['CLNSIG'])) != 1:\n",
    "            c.loc[c.POS == i, \"CONFLICTING\"] = 1      \n",
    "        else:\n",
    "            continue\n",
    "    return c"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "surprising-ballet",
   "metadata": {},
   "source": [
    "Our next step is to use both the functions we have created\n",
    "* First function 'get_chrom' is to get all the chromosomes dataframes with multiple variation entries\n",
    "* The second function is to assign if the categories present in every dataframe hase more than one values or not."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "prerequisite-disposal",
   "metadata": {},
   "source": [
    "We will use python exec function to create different variable names, and we will loop throuch the chromosome list we creaated to get the names. We will create individual dataframe for each chromosome."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "disabled-final",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Chrom 1 created ...\n",
      "Chrom 2 created ...\n",
      "Chrom 3 created ...\n",
      "Chrom 4 created ...\n",
      "Chrom 5 created ...\n",
      "Chrom 6 created ...\n",
      "Chrom 7 created ...\n",
      "Chrom 8 created ...\n",
      "Chrom 9 created ...\n",
      "Chrom 10 created ...\n",
      "Chrom 11 created ...\n",
      "Chrom 12 created ...\n",
      "Chrom 13 created ...\n",
      "Chrom 14 created ...\n",
      "Chrom 15 created ...\n",
      "Chrom 16 created ...\n",
      "Chrom 17 created ...\n",
      "Chrom 18 created ...\n",
      "Chrom 19 created ...\n",
      "Chrom 20 created ...\n",
      "Chrom 21 created ...\n",
      "Chrom 22 created ...\n",
      "Chrom 23 created ...\n",
      "Chrom X created ...\n",
      "Chrom Y created ...\n",
      "Chrom MT created ...\n",
      "Done\n"
     ]
    }
   ],
   "source": [
    "for i in chrom_list:\n",
    "    exec(f'c_{i} = get_chrom(df_1, i)')\n",
    "    print(f'Chrom {i} created ...')\n",
    "print('Done')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "sealed-polls",
   "metadata": {},
   "outputs": [],
   "source": [
    "new_chrom_list = [c_1, c_2, c_3, c_4, c_5, c_6, c_7, c_8, c_9, c_10,\n",
    "              c_11, c_12, c_13, c_14, c_15, c_16, c_17, c_18, c_19, c_20,\n",
    "              c_21, c_22, c_23, c_X, c_Y, c_MT ]\n",
    "\n",
    "new_chrom_str = ['c_1', 'c_2', 'c_3', 'c_4', 'c_5', 'c_6', 'c_7', 'c_8', 'c_9', 'c_10',\n",
    "              'c_11', 'c_12', 'c_13', 'c_14', 'c_15', 'c_16', 'c_17', 'c_18', 'c_19', 'c_20',\n",
    "              'c_21', 'c_22', 'c_23', 'c_X', 'c_Y', 'c_MT' ]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "seven-phase",
   "metadata": {},
   "source": [
    "We will use python zip function to apply the values from the list we have created above ie new_chrom_list(which has all the name of dataframes) and new_chrom_str(which has all the names of dataframe stroed as strings). We will use for loop and create individual dataframe for each chromosome using conflicting function."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "alien-retail",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "c_1 created ...\n",
      "c_2 created ...\n",
      "c_3 created ...\n",
      "c_4 created ...\n",
      "c_5 created ...\n",
      "c_6 created ...\n",
      "c_7 created ...\n",
      "c_8 created ...\n",
      "c_9 created ...\n",
      "c_10 created ...\n",
      "c_11 created ...\n",
      "c_12 created ...\n",
      "c_13 created ...\n",
      "c_14 created ...\n",
      "c_15 created ...\n",
      "c_16 created ...\n",
      "c_17 created ...\n",
      "c_18 created ...\n",
      "c_19 created ...\n",
      "c_20 created ...\n",
      "c_21 created ...\n",
      "c_22 created ...\n",
      "c_23 created ...\n",
      "c_X created ...\n",
      "c_Y created ...\n",
      "c_MT created ...\n",
      "Done\n"
     ]
    }
   ],
   "source": [
    "for i,j in zip(new_chrom_str, new_chrom_list):\n",
    "    exec(f'{i}_final = conflicting(j)')\n",
    "    print(f'{i} created ...')\n",
    "print('Done')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "interested-sample",
   "metadata": {},
   "source": [
    "Our final step is to concatinate all the dataframe we have created using concat function in pandas."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "tested-microphone",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_df =[c_1_final, c_2_final, c_3_final, c_4_final, c_5_final, c_6_final, c_7_final, c_8_final, c_9_final, c_10_final,\n",
    "         c_11_final, c_12_final, c_13_final, c_14_final, c_15_final, c_16_final, c_17_final, c_18_final, c_19_final, c_20_final, \n",
    "         c_21_final, c_22_final, c_23_final, c_X_final,c_Y_final,c_MT_final]\n",
    "final_df = pd.concat(all_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "infrared-puppy",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT_1</th>\n",
       "      <th>ALLELEID</th>\n",
       "      <th>CLNDN</th>\n",
       "      <th>CLNREVSTAT</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>CLNVC</th>\n",
       "      <th>MC</th>\n",
       "      <th>ORIGIN</th>\n",
       "      <th>numalt</th>\n",
       "      <th>altlen_1</th>\n",
       "      <th>is_snp</th>\n",
       "      <th>CONFLICTING</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>1</td>\n",
       "      <td>871143.0</td>\n",
       "      <td>1127328</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1110870</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001627|intron_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>1</td>\n",
       "      <td>871143.0</td>\n",
       "      <td>1158587</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>1131742</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001627|intron_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>1</td>\n",
       "      <td>871215.0</td>\n",
       "      <td>1170011</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1153710</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001819|synonymous_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>1</td>\n",
       "      <td>871215.0</td>\n",
       "      <td>1169806</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1153711</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001819|synonymous_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>1</td>\n",
       "      <td>874456.0</td>\n",
       "      <td>1122639</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1110873</td>\n",
       "      <td>not_provided</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>960019</th>\n",
       "      <td>MT</td>\n",
       "      <td>15852.0</td>\n",
       "      <td>693968</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>680858</td>\n",
       "      <td>Leigh_syndrome</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>960020</th>\n",
       "      <td>MT</td>\n",
       "      <td>15860.0</td>\n",
       "      <td>693970</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>680859</td>\n",
       "      <td>Leigh_syndrome</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>960021</th>\n",
       "      <td>MT</td>\n",
       "      <td>15860.0</td>\n",
       "      <td>693969</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>680860</td>\n",
       "      <td>Leigh_syndrome</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>960024</th>\n",
       "      <td>MT</td>\n",
       "      <td>15884.0</td>\n",
       "      <td>143900</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>153625</td>\n",
       "      <td>Neoplasm_of_ovary|Leigh_syndrome</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>960025</th>\n",
       "      <td>MT</td>\n",
       "      <td>15884.0</td>\n",
       "      <td>252455</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>247210</td>\n",
       "      <td>Leigh_syndrome|not_specified</td>\n",
       "      <td>criteria_provided</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>156822 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM       POS       ID REF ALT_1  ALLELEID  \\\n",
       "34         1  871143.0  1127328   G     A   1110870   \n",
       "35         1  871143.0  1158587   G     T   1131742   \n",
       "46         1  871215.0  1170011   C     G   1153710   \n",
       "47         1  871215.0  1169806   C     T   1153711   \n",
       "66         1  874456.0  1122639   G     A   1110873   \n",
       "...      ...       ...      ...  ..   ...       ...   \n",
       "960019    MT   15852.0   693968   T     G    680858   \n",
       "960020    MT   15860.0   693970   A     G    680859   \n",
       "960021    MT   15860.0   693969   A     T    680860   \n",
       "960024    MT   15884.0   143900   G     A    153625   \n",
       "960025    MT   15884.0   252455   G     C    247210   \n",
       "\n",
       "                                   CLNDN         CLNREVSTAT  \\\n",
       "34                          not_provided  criteria_provided   \n",
       "35                          not_provided  criteria_provided   \n",
       "46                          not_provided  criteria_provided   \n",
       "47                          not_provided  criteria_provided   \n",
       "66                          not_provided  criteria_provided   \n",
       "...                                  ...                ...   \n",
       "960019                    Leigh_syndrome  criteria_provided   \n",
       "960020                    Leigh_syndrome  criteria_provided   \n",
       "960021                    Leigh_syndrome  criteria_provided   \n",
       "960024  Neoplasm_of_ovary|Leigh_syndrome  criteria_provided   \n",
       "960025      Leigh_syndrome|not_specified  criteria_provided   \n",
       "\n",
       "                        CLNSIG                      CLNVC  \\\n",
       "34                      Benign  single_nucleotide_variant   \n",
       "35                      Benign  single_nucleotide_variant   \n",
       "46                      Benign  single_nucleotide_variant   \n",
       "47                      Benign  single_nucleotide_variant   \n",
       "66                      Benign  single_nucleotide_variant   \n",
       "...                        ...                        ...   \n",
       "960019  Uncertain_significance  single_nucleotide_variant   \n",
       "960020                  Benign  single_nucleotide_variant   \n",
       "960021                  Benign  single_nucleotide_variant   \n",
       "960024                  Benign  single_nucleotide_variant   \n",
       "960025                  Benign  single_nucleotide_variant   \n",
       "\n",
       "                                   MC  ORIGIN  numalt  altlen_1  is_snp  \\\n",
       "34          SO:0001627|intron_variant     1.0       1         0    True   \n",
       "35          SO:0001627|intron_variant     1.0       1         0    True   \n",
       "46      SO:0001819|synonymous_variant     1.0       1         0    True   \n",
       "47      SO:0001819|synonymous_variant     1.0       1         0    True   \n",
       "66        SO:0001583|missense_variant     1.0       1         0    True   \n",
       "...                               ...     ...     ...       ...     ...   \n",
       "960019    SO:0001583|missense_variant     1.0       1         0    True   \n",
       "960020    SO:0001583|missense_variant     1.0       1         0    True   \n",
       "960021    SO:0001583|missense_variant     1.0       1         0    True   \n",
       "960024    SO:0001583|missense_variant     1.0       1         0    True   \n",
       "960025    SO:0001583|missense_variant     1.0       1         0    True   \n",
       "\n",
       "        CONFLICTING  \n",
       "34                0  \n",
       "35                0  \n",
       "46                0  \n",
       "47                0  \n",
       "66                1  \n",
       "...             ...  \n",
       "960019            1  \n",
       "960020            0  \n",
       "960021            0  \n",
       "960024            0  \n",
       "960025            0  \n",
       "\n",
       "[156822 rows x 16 columns]"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "informational-surfing",
   "metadata": {},
   "source": [
    "We have our final dataset where we can perform statistical analysis and create classification model effectively."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "modern-bacteria",
   "metadata": {},
   "outputs": [],
   "source": [
    "final_df.to_csv('final_df.csv',index=False)# converting final_df to csv file for further process"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "designed-deployment",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
